### PR TITLE
Update cdb_tablemetadata when a column is casted

### LINKED ISSUE #2203: api does not update cdb_tablemedata when a column is casted
it only invalidates varnish: 

https://github.com/CartoDB/cartodb/blob/2182-geometry_types_frontend/app/models/table.rb#L1045

``update_cdb_tablemetadata`` should be called

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #2203.

@javisantana PTAL. Is there a way to test that this is working through the UI?

### CODE REVIEW

File: spec/models/table_spec.rb [hunk 1]
```diff
@@ -70,6 +70,7 @@ def @data_import.data_source=(filepath)
       table         = Table.new
       table.user_id = @user.id
       table.name    = 'layergroup'
+      table.name.should eq 'layergroup_t'
```
Comment (alonsogarciapablo) [inline spec/models/table_spec.rb]: (bonus)

File: spec/models/table_spec.rb [hunk 2]
```diff
@@ -737,6 +738,19 @@ def @data_import.data_source=(filepath)
...
 
+    it "should invalidate varnish cache after modifying a column" do
+      table = create_table(:user_id => @user.id)
+      table.expects(:invalidate_varnish_cache)
+      table.modify_column!(:name => "name", :type => "number")
+    end
+
+    it "should update public.cdb_tablemetadata after modifying a column" do
+      table = create_table(:user_id => @user.id)
+      table.expects(:update_cdb_tablemetadata).once
```
Comment (alonsogarciapablo) [inline spec/models/table_spec.rb]: Perhaps there's a better way to test this?

**Comment by javisantana** [comment]
it's better than @rafatower review this

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2514/))

**Comment by rafatower** [comment]
ok with it

**Comment by alonsogarciapablo** [comment]
Great. Is there a way to test this through the UI?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2516/))

**Comment by alonsogarciapablo** [comment]
I change the type of a column and this popped up in the console. I guess that's a good validation :)

```
16:36:43 web.1    | [pool] Found a connection for localhost:cartodb_dev_user_005a6815-9a71-4107-8e92-1e71aaae467a_db:postgres (2)
16:36:43 web.1    | (0.006245s) 
16:36:43 web.1    |       SELECT CDB_TableMetadataTouch('36929')
```

**Pull request merged**